steps/pytorch/architectures/unet.py

```diff
@@ -44,38 +53,45 @@ def _up_convs(self):
 
     def _down_pools(self):
         down_pools = []
+        padding = get_downsample_pad(stride=self.pool_stride, kernel=self.pool_kernel)
         for _ in range(self.repeat_blocks):
-            down_pools.append(nn.Sequential(nn.ConstantPad2d(1, 0),
-                                            nn.MaxPool2d(kernel_size=(self.pool_kernel, self.pool_kernel),
-                                                         stride=self.pool_stride)))
+            down_pools.append(nn.MaxPool2d(kernel_size=self.pool_kernel,
+                                           stride=self.pool_stride,
+                                           padding=padding))
         return nn.ModuleList(down_pools)
 
     def _up_samples(self):
         up_samples = []
+        kernel_scale = self.kernel_scale
+        stride = self.pool_stride
+        kernel_size = kernel_scale * stride
+        padding, output_padding = get_upsample_pad(stride=stride, kernel=kernel_size)
         for i in range(self.repeat_blocks):
             in_channels = int(self.n_filters * 2 ** (i + 2))
             out_channels = int(self.n_filters * 2 ** (i + 1))
             up_samples.append(nn.ConvTranspose2d(in_channels=in_channels,
                                                  out_channels=out_channels,
-                                                 kernel_size=2,
-                                                 stride=2,
-                                                 padding=0,
-                                                 output_padding=0,
+                                                 kernel_size=kernel_size,
+                                                 stride=stride,
+                                                 padding=padding,
+                                                 output_padding=output_padding,
                                                  bias=False
                                                  ))
         return nn.ModuleList(up_samples)
 
     def _input_block(self):
+        stride = 1
```

Comment:
@apyskir Should we parametrize this as well?

Reply:
The UNet architecture we use assumes, that convolutional layers don't change the size of the image. Therefore stride = 1 is hard-coded (as it was in DSB code). I can make a global variable CONV_STRIDE = 1 and use it in all cases, when I use stride = 1 now - what do you think? @kamil-kaczmarek @jakubczakon

Reply:
@apyskir I think that it is potentially safer to have it defined in one place and dump it/change it if need be. So yes I would have a variable conv_stride in unet definition .I think all parameters should be defined inside of the Unet definition not outside of it and hence I don't like global CONV_STRIDE

Reply:
@jakubczakon @kamil-kaczmarek Done. Please review.